File: C programming/Assignment/Assignment5/type4/perfect.c
#include <stdio.h>

int perfect(int num);
void main(){
    int num;
    printf("Enter the number:");
    scanf("%d",&num);
    if(perfect(num)==1)
        printf("perfect number");
    else    
        printf("not perfect number");

}

int perfect(int num){
    int perfect,sum=0,i=1;

    perfect = num;

    while(i<=num/2){  //need to check only half of the number 
        if(num%i==0){
            sum = sum + i;
        }
        i++;
    }
    if(perfect==sum)
        return 1;

    return 0;
}

File: Java_programming/basic_java/assignment3/perfect.java
package assignment3;

public class perfect {

	public static void main(String[] args) {
		int num=6,perfect,sum=0,i=1;
//	    printf("Enter the number:");
//	    scanf("%d",&num);

	    perfect = num;

	    while(i<=num/2){  //need to check only half of the number 
	        if(num%i==0){
	            sum = sum + i;
	        }
	        i++;
	    }
	    

	    if(perfect==sum){
	        System.out.printf("number is perfect\n");
	    }
	    else{
	    	System.out.printf("number is not perfect\n");
	    }

	}

}
